infra/modules/common_environment/variables.tf

```diff
@@ -0,0 +1,58 @@
+#---------#
+# General #
+#---------#
+
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "environment" {
+  type = object({
+    prefix          = string
+    env_short       = string
+    location        = string
+    domain          = optional(string)
+    app_name        = string
+    instance_number = string
+  })
+
+  validation {
+    condition     = length("${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : replace(var.environment.domain, "-", "")}${var.environment.app_name}-apim-${var.environment.instance_number}") <= 50
+    error_message = "Azure API Management name must contain between 1 and 50 characters. Current value is \"${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : var.environment.domain}${var.environment.app_name}-apim-${var.environment.instance_number}\""
+  }
+
+  description = "Values which are used to generate resource names and location short names. They are all mandatory except for domain, which should not be used only in the case of a resource used by multiple domains."
+}
+
+variable "test_enable" {
+  type        = bool
+  description = "Flag to enable testing resources"
+  default     = false
+}
+
+# Networking variables
+
+variable "virtual_network_cidr" {
+  type        = string
+  description = "CIDR block for the virtual network"
```

Comment:
Can we consider a default one?

Reply:
You only have collisions if two vnets with same cidr are paired, but you can have multiple vnets with the same block addresses, also in the same rg!

Reply:
ok! So i think we can use `10.0.0.0/16` as default